### PR TITLE
Add Silicon Labs Multiprotocol container

### PR DESCRIPTION
**Pull request by agners**
Add Silicon Labs Multiprotocol container to development add-ons.
Currently this container requires special hardware/firmware to be
useful. Adding it here makes it available to developers for now.

### CODE REVIEW

File: silabs-multiprotocol/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Home Assistant Add-on: SiliconLabs Zigbee/OpenThread Multiprotocol Add-on
+
+Zigbee/OpenThread Multiprotocol container for Silicon Labs based radios.
+
+**NOTE:** Requires Silicon Labs based radio with RCP Multi-PAN firmware flashed!
+
+![Supports aarch64 Architecture][aarch64-shield] ![Supports armv7 Architecture][armv7-shield]
```
Comment (ludeeus) [inline silabs-multiprotocol/README.md:7]: aarch64 is disabled in build.yaml and config.yaml

File: silabs-multiprotocol/config.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+description: Zigbee and OpenThread multiprotocol add-on
+url: https://github.com/home-assistant/hassio-addons/tree/master/silabs-multiprotocol
+arch:
+  - armv7
+# Currently zigbeed crashes with *** stack smashing detected ***
+#  - aarch64
+hassio_api: true
+privileged:
+  - IPC_LOCK
+image: ghcr.io/home-assistant/{arch}-addon-silabs-multiprotocol
```
Comment (ludeeus) [inline silabs-multiprotocol/config.yaml:13]: You need to add shadow repository for this <https://github.com/home-assistant/supervisor/blob/main/build.yaml#L2>

File: silabs-multiprotocol/config.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+init: false
+options:
+  device: null
+  cpcd_trace: false
+  socat_trace: false
+ports:
+  9999/tcp: null
+ports_description:
```
Comment (ludeeus) [inline silabs-multiprotocol/config.yaml:21]: translations/en.yaml? <https://developers.home-assistant.io/docs/add-ons/configuration#add-on-translations>

File: silabs-multiprotocol/config.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+version: 0.1.0
+slug: silabs_multiprotocol
+name: Silicon Labs Multiprotocol
+description: Zigbee and OpenThread multiprotocol add-on
+url: https://github.com/home-assistant/hassio-addons/tree/master/silabs-multiprotocol
```
Comment (ludeeus) [inline silabs-multiprotocol/config.yaml:5]:
```suggestion
url: https://github.com/home-assistant/addons-development/tree/master/silabs-multiprotocol
```

File: silabs-multiprotocol/config.yaml (hunk 2 above)
Comment (agners) [reply silabs-multiprotocol/config.yaml:13]: Oh I see, the other addons are still on Docker Hub then?

I don't see a reason why we should make this special, so how about:

```suggestion
image: homeassistant/{arch}-addon-silabs-multiprotocol
```

File: silabs-multiprotocol/README.md (hunk 1 above)
Comment (agners) [reply silabs-multiprotocol/README.md:7]: Yeah I mean I can build it but zigbeed crashes immediately. I'd rather prefer to use the working armv7 version right now on aarch64 :sweat_smile: 

Will remove the aarch64 badge for now.

File: silabs-multiprotocol/config.yaml (hunk 2 above)
Comment (ludeeus) [reply silabs-multiprotocol/config.yaml:13]: That works :+1: 
we can also remove the ghcr.io login part of the CI BTW, it's not used so no need to be there

Comment (agners) [reply silabs-multiprotocol/config.yaml:13]: Ok, I'll do that in a separate PR. It is also not needed in addons repo afaict.

File: silabs-multiprotocol/rootfs/etc/services.d/cpcd/run [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/with-contenv bashio
+# ==============================================================================
+# Start Co-Processor Communication Daemon (CPCd)
+# ==============================================================================
+bashio::log.info "Starting cpcd..."
+s6-notifyoncheck -s 300 -w 300 "/usr/bin/stdbuf" -o0 /usr/local/bin/cpcd
+bashio::log.info "Cleaning up after cpcd..."
+rm -r /dev/shm/cpcd
```
Comment (pvizeli) [inline silabs-multiprotocol/rootfs/etc/services.d/cpcd/run:8]: You can run that as normal service and use the finish script to clean it up?

File: silabs-multiprotocol/build.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+build_from:
+#  aarch64: ghcr.io/home-assistant/aarch64-base-debian:bullseye
+  armv7: ghcr.io/home-assistant/armv7-base-debian:bullseye
```
Comment (pvizeli) [inline silabs-multiprotocol/build.yaml:3]: as we build the binary, alpine would not work?

File: silabs-multiprotocol/rootfs/etc/services.d/cpcd/run (hunk 5 above)
Comment (agners) [reply silabs-multiprotocol/rootfs/etc/services.d/cpcd/run:8]: What do you mean by normal?

`s6-notifyoncheck` I need so the service is only defined as up when the cpcd is really ready to be talked to.

But I agree, cpcd clean-up can be moved to a finish script. Will do that.

File: silabs-multiprotocol/build.yaml (hunk 6 above)
Comment (agners) [reply silabs-multiprotocol/build.yaml:3]: Currently the `zigbeed` binary is copied from  their SDK, so no.

But I plan to build it myself at one point. It requires their IDE afaik, so it probably will be injected as binary still. I'll keep it in mind then to try using alpine.

**Review by pvizeli — APPROVED** [review]
Maybe you should add `host_ipc: false` as we speak about, but open to you

**Pull request merged**